lib/tools/broccoli/builder.ts

```diff
@@ -10,78 +11,86 @@ let vinylFilterSince = require("vinyl-filter-since");
 let through = require("through2");
 
 export class Builder implements IBroccoliBuilder {
-   private nodeModules: any = {};
-
   constructor(private $fs: IFileSystem,
     private $nodeModulesTree: INodeModulesTree,
+    private $projectData: IProjectData,
     private $projectDataService: IProjectDataService,
     private $injector: IInjector,
     private $logger: ILogger) { }
 
-  public prepareNodeModules(absoluteOutputPath: string, projectDir: string, platform: string, lastModifiedTime?: Date): IFuture<void> {
+    public getChangedNodeModules(absoluteOutputPath: string, platform: string, lastModifiedTime?: Date): IFuture<any> {
+        return (() => {
+            let projectDir = this.$projectData.projectDir;
+            let isNodeModulesModified = false;
```

Comment:
Is this the correct place for this var? Shouldn't it be local to the pipe callback where it is used?